### DISCUSSION TITLE
Rich Live Table performance

### DISCUSSION THREAD

**Discussion by xwjiang2010** [Textualize/rich#2841] [Q&A]
Hi!
I am trying to integrate [Ray Tune library](https://github.com/ray-project/ray) with Rich. 
Attached is what I have right now using Live table. However I noticed there is a noticeable 10% performance drop with some artificial workload. 
I think the reason is that the table rendering which is refreshed 10 times per second (in the video) competes with other tasks on the main thread for CPU time. 
Want to understand if this is expected and does anyone have any suggestions for that?

Thanks so much in advance!

https://user-images.githubusercontent.com/87673679/222604062-b9417060-cf47-4cd2-a255-5189cae93993.mp4

**Comment by willmcgugan** [comment]
You can change the refresh rate of Live to be a little lower, which would reduce CPU load. However, I see you are using macOS Terminal, which is known to be slow. You might want to install iTerm2 on macOS.

Alternatively, you could use [Textual](https://github.com/Textualize/textual) to create an interface like this. Textual is designed for dynamic interfaces and will likely be more performant.
